plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -729,6 +735,87 @@ void *restore_bo_contents(void *_thread_data)
 	return NULL;
 };
 
+int check_hsakmt_shared_mem(uint64_t *shared_mem_size, uint32_t *shared_mem_magic)
+{
+	int ret, fd;
+	struct stat st;
+	uint32_t *shared_mem_ptr;
+
+	ret = stat(HSAKMT_SHM_PATH, &st);
+	if (ret) {
+		*shared_mem_size = 0;
+		return 0;
+	}
+
+	fd = shm_open(HSAKMT_SHM, O_RDONLY, 0);
+	if (fd < 0) {
+		*shared_mem_size = 0;
+		return 0;
+	}
+
+	shared_mem_ptr = mmap(0, st.st_size, PROT_READ, MAP_SHARED, fd, 0);
+	if (shared_mem_ptr == MAP_FAILED) {
+		pr_perror("amdgpu_plugin: Failed to mmap shared mem %s\n", HSAKMT_SHM);
+		close(fd);
+		return -EACCES;
+	}
+
+	*shared_mem_size = st.st_size;
+
+	/* First 4 bytes of shared file is the magic */
+	memcpy(shared_mem_magic, shared_mem_ptr, sizeof(*shared_mem_magic));
```

Comment:
You can do this with a normal assignment:

    *shared_mem_magic = *shared_mem_ptr;

But that said, is it really worth doing this whole mmap dance for copying a single value? Why not just use a read system call?

Reply:
ACK